### ISSUE TITLE
Find -print0 false negative

### ISSUE THREAD

**Issue by jpaugh** [koalaman/shellcheck#173]
## The problem

The following code snippet produces an error,  but it should not:

``` sh
find -not -type d -print0 | xargs --null ls
```

The error message is

```
SC2038 Use either 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow for non-alphanumeric filenames.
```

This error was encountered while using http://www.shellcheck.net/. 
## Evaluation

This happens because `xargs` has been passed the `--null` option instead of the `-0` option. The code can easily (I think) be changed check for both forms to avoid this issue. The relevant portion of code is [here](https://github.com/koalaman/shellcheck/blob/master/ShellCheck/Analytics.hs#L491).
## The bigger picture

Perhaps this lint-check should be split into two?
- One to ensure `find` uses `-print0` before a pipe, and
- The other to ensure `xargs` is not mismatched with the `find` command: either `find .. | xargs ..` or `find .. -print0 | xargs -0` would be accepted&nbsp;&mdash; although the former would still trigger a lint warning from the previous bullet, of course.

This would allow greater flexibility in the future:
- Ensuring `find .. -print0` is used when piping to other, null-aware programs (such as `cpio`)
- Ensuring other commands (such as `grep`) which can output null-delimited format do so whenever possible (i.e. in the same situations as `find`).
- Ensuring `xargs` is not mismatched to any other command which can output null-delimited lines.
- Ensuring other commands (such as `cpio`) which can read null-delimited format are not mismatched, parallel to the `xargs` case.
